src/main/java/org/wso2/mi/tool/connector/tools/generator/grpc/GRPCConnectorGenerator.java

```diff
@@ -0,0 +1,211 @@
+            List<DescriptorProtos.ServiceDescriptorProto> serviceList = fileProto.getServiceList();
+            for (DescriptorProtos.ServiceDescriptorProto service : serviceList) {
+                String serviceName = service.getName();
+                String resolvedConnectorName = serviceName.toLowerCase().replace(" ", "");
+                updateConnectorMetaInfo(context, serviceName, resolvedConnectorName);
+
+                List<DescriptorProtos.MethodDescriptorProto> methodList = service.getMethodList();
+                Map<String, RPCService.RPCCall> rcpMap = new HashMap<>();
+                for (DescriptorProtos.MethodDescriptorProto method : methodList) {
+                    if (method.getServerStreaming() || method.getClientStreaming()) {
+                        LOG.warn(ErrorMessages.GRPC_CONNECTOR_102.format(method.getName()));
+                    } else {
+                        populateRPCcall(fileProto, messageTypeMap, rcpMap, method);
+                    }
+                }
+                RPCService rpcService = new RPCService.Builder()
+                        .serviceName(serviceName)
+                        .addRpcCall(rcpMap)
+                        .build();
+                context.put(SERVICE, rpcService);
+            }
+            break;
+        }
+        return context;
+    }
+
+    private static void populateRPCcall(DescriptorProtos.FileDescriptorProto fileProto,
+                                        Map<String, DescriptorProtos.DescriptorProto> messageTypeMap,
+                                        Map<String, RPCService.RPCCall> rcpMap,
+                                        DescriptorProtos.MethodDescriptorProto method) {
+        String methodName = method.getName();
+        String inputType = method.getInputType();
+        String packageName = fileProto.getPackage();
+        String inputTypeName = getTypeName(inputType, packageName);
+        Map<String, DescriptorProtos.FieldDescriptorProto> inputFields = fieldMap(messageTypeMap, inputTypeName);
+        String outputType = method.getOutputType();
+        String outputName = getTypeName(outputType, packageName);
+        Map<String, DescriptorProtos.FieldDescriptorProto> outputFields = fieldMap(messageTypeMap,
+                getTypeName(outputType, packageName));
+        RPCService.RPCCall.RPCCallBuilder callBuilder = new RPCService.RPCCall.RPCCallBuilder()
+                .rpcCallName(methodName)
+                .inputName(inputTypeName)
+                .outputName(outputName)
+                .addInputParam(inputFields)
+                .addOutputParam(outputFields);
+        RPCService.RPCCall rcpCall = callBuilder.build();
+        rcpMap.put(methodName, rcpCall);
+    }
+
+    private static void updateConnectorMetaInfo(VelocityContext context, String serviceName, String resolvedConnectorName) {
+        String artifactId = "org.wso2.mi.connector." + resolvedConnectorName;
+        context.put("artifactId", artifactId);
```

Comment:
We could make things like these constants. No mandatory though. These and when you compare two strings eg: `if (idlFile.endsWith(".proto")) {`